lib/routes/signin-codes.js

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const isA = require('joi')
+const validators = require('./validators')
+
+const METRICS_CONTEXT_SCHEMA = require('../metrics/context').requiredSchema
+
+module.exports = (log, db, config, customs) => {
+  const CODE_LENGTH = Math.ceil(4 * (config.signinCodeSize / 3))
+
+  return [
+    {
+      method: 'POST',
+      path: '/signinCodes/consume',
+      config: {
+        validate: {
+          payload: {
+            code: isA.string().regex(validators.URL_SAFE_BASE_64).length(CODE_LENGTH).required(),
```

Comment:
You make an excellent point. With the code as it is, no they can't. I could swap out `length` for `min`, as it seems inconceivable that we'd ever want to shorten the code. Or I could just ditch length validation altogether and leave it up to the db to validate when we go to look it up. Which do you think is better?